base/sim/FairMCApplication.cxx

```diff
@@ -579,13 +568,13 @@ void FairMCApplication::Stepping()
     // In any case call the ProcessHits function for this specific detector.
     Int_t copyNo;
     Int_t id = fMC->CurrentVolID(copyNo);
-    Bool_t InMap = kFALSE;
-    fDisVol = 0;
-    fDisDet = 0;
     Int_t fCopyNo = 0;
-    fVolIter = fVolMap.find(id);
```

Comment:
What is the time needed to create the iterator? The stepping is the function which is called most during the transport simulation so it could well be that the iterator was a class member. 
Please  do a large scale transport simulation to  check the effect of the change on the runtime.
@MohammadAlTurany, what do you mean?

Reply:
Well, take a look at the generated assembly:

https://godbolt.org/z/87fdfnenW

1. Instanatiating a `std` iterator seems to be as simple as setting one qword to zero.
2. The assembly with the local variable looks longer and more convoluted to me. I haven't taken the time to analyze it fully, but it looks like it's very much the same, except for having to store the results in the member variable.

Reply:
The transport simulation runtime is comparable, no substantial time differences.